proposals/vendir/001-lazy-synching-on-stable-config/README.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Lazy Synching"
+authors: [ "Fritz Duchardt <fritz.duchardt.ext@gec.io>" ]
+status: "draft"
+approvers: [ ]
+---
+
+# Make vendir lazy: don't syncing if the config has not changed
+
+## Problem Statement
+We have a rollout mechanism that heavily relies on vendir to pull in upstream workloads from Helm Charts, OCI Images and Git Repositories. Most of these workloads have fixed versions and don't change unless their corresponding vendir.yaml is modified. Yet, currently vendir constrains us to sync everything on every run, which slows down the process unnecessarily.
+
+## Proposal
+Add an additional config option to run sync only if the config was changed after the last sync. Users can activate this feature for all workloads they deem stable.
+
+### Goals
+Users benefit from quicker syncing process, if they use vendir to pull in stable release versions.
+
+### Specification / Use Cases
+The feature can be activated in the `vendir.yml` at `contents` level, e.g.:
+
+```
+apiVersion: vendir.k14s.io/v1alpha1
+kind: Config
+directories:
+- path: vendor
+  contents:
+  - path: custom-repo-custom-version
+    lazy: true
+    helmChart:
+      name: contour
+      version: "7.10.1"
+      repository:
+        url: https://charts.bitnami.com/bitnami
+```
+
+Lazily `contents` are synced under two conditions:
+- When their corresponding config section has changed. The mechanism works in a generic fashion for all repository types and reacts to any change to the `contents` config section. 
+- When the path of the parent `directory` has changed.
+
+To track changes to the config, the vendir.lock.yml is amended with a hash value that represents the state of a `contents` config section at the last sync, e.g.:
+```
+kind: LockConfig
+apiVersion: vendir.k14s.io/v1alpha1
+directories:
+- path: vendor
+  contents:
+  - path: custom-repo-custom-version
+    hash: e8a5d1511f2eb22b160bb849e5c8be39da1c4ffa5fd56ded71ff095a8b24720b
```

Comment:
let's rename `hash` (too generic) field to something like `configDigest` -- i'm thinking in future we may need to implement `contentDigest` for ensuring content stays exactly as expected.